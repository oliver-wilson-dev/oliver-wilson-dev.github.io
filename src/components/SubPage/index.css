@import "../../styles/shared.css";

.subPage {
    max-width: var(--pageContentMaxWidth);
    padding: 30px;
}