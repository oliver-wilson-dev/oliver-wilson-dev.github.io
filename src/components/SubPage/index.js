import React from 'react';
import PropTypes from 'prop-types';

import styles from './index.css';

const SubPage = ({ children }) => (
  <div className={styles.subPage}>{children}</div>
);

SubPage.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default SubPage;
