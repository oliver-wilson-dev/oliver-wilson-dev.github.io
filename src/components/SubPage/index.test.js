import React from 'react';
import { shallow } from 'enzyme';
import SubPage from '.';

const render = (props = {}) => shallow(<SubPage {...props}><h1>hello world</h1></SubPage>);

describe('SubPage', () => {
  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when provided with an additionalStyles prop', () => {
    it('should apply that class to the wrapper div', () => {
      const mockClass = 'test-additional-styles';
      const component = render({ additionalStyles: mockClass });

      expect(component.find(`.${mockClass}`).exists()).toBe(true);
    });
  });
});
