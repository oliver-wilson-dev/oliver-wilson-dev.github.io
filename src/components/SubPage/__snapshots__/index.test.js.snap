// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubPage should render correctly 1`] = `
<div
  className="subPage"
>
  <h1>
    hello world
  </h1>
</div>
`;
